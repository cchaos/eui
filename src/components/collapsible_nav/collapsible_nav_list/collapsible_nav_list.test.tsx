import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import {
  EuiCollapsibleNavList,
  EuiCollapsibleNavListItemProps,
} from './collapsible_nav_list';

const someListItems: EuiCollapsibleNavListItemProps[] = [
  {
    label: 'Label with iconType',
    iconType: 'stop',
  },
  {
    label: 'Custom extra action',
    extraAction: {
      iconType: 'bell',
      alwaysShow: true,
    },
  },
  {
    label: 'Button with onClick',
    pinned: true,
    onClick: e => {
      console.log('Visualize clicked', e);
    },
  },
  {
    label: 'Link with href',
    href: '#',
  },
];

describe('EuiCollapsibleNavList', () => {
  test('is rendered', () => {
    const component = render(
      <EuiCollapsibleNavList listItems={someListItems} {...requiredProps} />
    );

    expect(component).toMatchSnapshot();
  });

  test('renders pinning options only if onPinClick is provided', () => {
    const component = render(
      <EuiCollapsibleNavList listItems={someListItems} onPinClick={() => {}} />
    );

    expect(component).toMatchSnapshot();
  });
});
