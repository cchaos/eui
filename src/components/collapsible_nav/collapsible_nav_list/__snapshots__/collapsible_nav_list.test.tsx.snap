// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCollapsibleNavList is rendered 1`] = `
<ul
  aria-label="aria-label"
  class="euiListGroup euiListGroup-maxWidthDefault euiCollapsibleNavList testClass1 testClass2"
  data-test-subj="test subject string"
>
  <li
    class="euiListGroupItem euiListGroupItem--small euiCollapsibleNavList__item"
  >
    <span
      class="euiListGroupItem__text"
    >
      <div
        class="euiListGroupItem__icon"
        data-euiicon-type="stop"
      />
      <span
        class="euiListGroupItem__label"
        title="Label with iconType"
      >
        Label with iconType
      </span>
    </span>
  </li>
  <li
    class="euiListGroupItem euiListGroupItem--small euiListGroupItem-hasExtraAction euiCollapsibleNavList__item"
  >
    <span
      class="euiListGroupItem__text"
    >
      <span
        class="euiListGroupItem__label"
        title="Custom extra action"
      >
        Custom extra action
      </span>
    </span>
    <button
      class="euiButtonIcon euiButtonIcon--primary euiListGroupItem__extraAction euiListGroupItem__extraAction-alwaysShow"
      type="button"
    >
      <div
        aria-hidden="true"
        class="euiButtonIcon__icon"
        data-euiicon-type="bell"
      />
    </button>
  </li>
  <li
    class="euiListGroupItem euiListGroupItem--small euiListGroupItem-isClickable euiCollapsibleNavList__item"
  >
    <button
      class="euiListGroupItem__button"
      type="button"
    >
      <span
        class="euiListGroupItem__label"
        title="Button with onClick"
      >
        Button with onClick
      </span>
    </button>
  </li>
  <li
    class="euiListGroupItem euiListGroupItem--small euiListGroupItem-isClickable euiCollapsibleNavList__item"
  >
    <a
      class="euiListGroupItem__button"
      href="#"
    >
      <span
        class="euiListGroupItem__label"
        title="Link with href"
      >
        Link with href
      </span>
    </a>
  </li>
</ul>
`;
